### ISSUE TITLE
Can't locate Win32/Job.pm

### ISSUE THREAD

**Issue by dkindlund** [dkindlund/honeyclient#178]
Hello, 

while running/testing the bootstrap_agent I get the following error message displayed in the console?  Any ideas?? All other tests have passed thus far.  I have installed via svn.  Below is version information.

Thanks for the help.

$ ~/honeyclient/bin/bootstrap_agent.sh
Starting up Agent - (Hit CTRL-C multiple times to exit.)
IP = 192.168.0.129

Pinging pingu.honeyclient.org [69.93.167.114] with 32 bytes of data:

Reply from 69.93.167.114: bytes=32 time=69ms TTL=128
Reply from 69.93.167.114: bytes=32 time=36ms TTL=128
Reply from 69.93.167.114: bytes=32 time=35ms TTL=128
Reply from 69.93.167.114: bytes=32 time=36ms TTL=128

Ping statistics for 69.93.167.114:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 35ms, Maximum = 69ms, Average = 44ms
At revision 1656.
Option: Collecting modified files
Option: Logging system events to C:\cygwin\tmp\realtime-changes.txt
Driver already loaded: CaptureProcessMonitor
Driver already loaded: CaptureRegistryMonitor
## Loaded filter driver: CaptureFileMonitor

2008-07-07 11:29:12 FATAL [HoneyClient::Agent::init](lib/HoneyClient/Agent.pm:365) - Can't locate Win32/Job.pm in @INC (@INC contains: lib /usr/lib/perl5/5.10/i686-cygwin /usr/lib/perl5/5.10 /usr/lib/perl5/site_perl/5.10/i686-cygwin /usr/lib/perl5/site_perl/5.10 /usr/lib/perl5/vendor_perl/5.10/i686-cygwin /usr/lib/perl5/vendor_perl/5.10 /usr/lib/perl5/vendor_perl/5.10 /usr/lib/perl5/site_perl/5.8 /usr/lib/perl5/vendor_perl/5.8 . lib) at lib/HoneyClient/Agent/Driver/Browser/IE.pm line 277.
BEGIN failed--compilation aborted at lib/HoneyClient/Agent/Driver/Browser/IE.pm line 277.
Compilation failed in require at (eval 159) line 2.
BEGIN failed--compilation aborted at (eval 159) line 2.
 at bin/StartAgent.pl line 21
2008-07-07 11:29:17 FATAL [HoneyClient::Agent::init](lib/HoneyClient/Agent.pm:365) - Can't locate Win32/Job.pm in @INC (@INC contains: lib /usr/lib/perl5/5.10/i686-cygwin /usr/lib/perl5/5.10 /usr/lib/perl5/site_perl/5.10/i686-cygwin /usr/lib/perl5/site_perl/5.10 /usr/lib/perl5/vendor_perl/5.10/i686-cygwin /usr/lib/perl5/vendor_perl/5.10 /usr/lib/perl5/vendor_perl/5.10 /usr/lib/perl5/site_perl/5.8 /usr/lib/perl5/vendor_perl/5.8 . lib) at lib/HoneyClient/Agent/Driver/Browser/IE.pm line 277.
BEGIN failed--compilation aborted at lib/HoneyClient/Agent/Driver/Browser/IE.pm line 277.
Compilation failed in require at (eval 159) line 2.
BEGIN failed--compilation aborted at (eval 159) line 2.
 at bin/StartAgent.pl line 21
2008-07-07 11:29:22 FATAL [HoneyClient::Agent::init](lib/HoneyClient/Agent.pm:365) - Can't locate Win32/Job.pm in @INC (@INC contains: lib /usr/lib/perl5/5.10/i686-cygwin /usr/lib/perl5/5.10 /usr/lib/perl5/site_perl/5.10/i686-cygwin /usr/lib/perl5/site_perl/5.10 /usr/lib/perl5/vendor_perl/5.10/i686-cygwin /usr/lib/perl5/vendor_perl/5.10 /usr/lib/perl5/vendor_perl/5.10 /usr/lib/perl5/site_perl/5.8 /usr/lib/perl5/vendor_perl/5.8 . lib) at lib/HoneyClient/Agent/Driver/Browser/IE.pm line 277.
BEGIN failed--compilation aborted at lib/HoneyClient/Agent/Driver/Browser/IE.pm line 277.
Compilation failed in require at (eval 159) line 2.
BEGIN failed--compilation aborted at (eval 159) line 2.
 at bin/StartAgent.pl line 21
2008-07-07 11:30:14 FATAL [HoneyClient::Agent::init](lib/HoneyClient/Agent.pm:365) - Can't locate Win32/Job.pm in @INC (@INC contains: lib /usr/lib/perl5/5.10/i686-cygwin /usr/lib/perl5/5.10 /usr/lib/perl5/site_perl/5.10/i686-cygwin /usr/lib/perl5/site_perl/5.10 /usr/lib/perl5/vendor_perl/5.10/i686-cygwin /usr/lib/perl5/vendor_perl/5.10 /usr/lib/perl5/vendor_perl/5.10 /usr/lib/perl5/site_perl/5.8 /usr/lib/perl5/vendor_perl/5.8 . lib) at lib/HoneyClient/Agent/Driver/Browser/IE.pm line 277.
BEGIN failed--compilation aborted at lib/HoneyClient/Agent/Driver/Browser/IE.pm line 277.
Compilation failed in require at (eval 159) line 2.
BEGIN failed--compilation aborted at (eval 159) line 2.
Caught a SIGINT at /usr/lib/perl5/5.10/i686-cygwin/Sys/Syslog.pm line 744

SVN INFO OUTPUT
$ svn info honeyclient/
Path: honeyclient
URL: svn://svn.honeyclient.org/honeyclient/tags/rel/1.0.2
Repository Root: svn://svn.honeyclient.org
Repository UUID: 143ac459-0e48-db11-92d1-000d614347cd
Revision: 1656
Node Kind: directory
Schedule: normal
Last Changed Author: xkovah
Last Changed Rev: 1339
Last Changed Date: 2008-03-07 12:41:10 -0500 (Fri, 07 Mar 2008)

**Comment by dkindlund** [comment]
Author: xkovah
I think this is a problem I anticipated happening because we've continued to make changes to the user guide, but not released a 1.0.3 version to match it.

If you checked out the 1.0.2 version, the Bundle::HoneyClient::Agent will not contain the Win32::Job package, as originally that was supposed to be installed via a cygwin perl-libwin32 package. However, as you can see that package is no longer listed as required on the UserGuide, because the current trunk version just installs it via Bundle::HoneyClient::Agent.

Therefor I would say you should try installing the perl-libwin32 cygwin package, and if that doesn't get rid of the error, try installing Win32::Job directly through CPAN. Then if the error persists, let us know. In the mean time we will work on getting 1.0.3 released.

Xeno

**Comment by dkindlund** [comment]
Author: kindlund
Thanks, Xeno.

To clarify, here's how you would install Win32::Job manually.
1. Get into a cygwin shell
2. Type "cpan".
3. From the cpan> prompt, type:

{{{
cpan> install Win32::Job
}}}

Hope this helps,

--Darien

**Comment by dkindlund** [comment]
Author: xkovah
Further investigation revealed that this is actually an issue with Win32::Job requiring g++ to compile. I had g++ already installed on my test VM therefore I wasn't aware of this prereq until I tried it on a VM with Cygwin wiped off it. I have updated the UserGuide to add "gcc-g++" as an additional package which must be installed via Cygwin.

So either the installing the Cygwin binary package I previously mentioned, or installing gcc-g++ and then Win32::Job should work.

Xeno
